src/tools/data/datagen/testall.C

```diff
@@ -351,6 +353,7 @@ build_rect2d(DBfile * dbfile, int size)
     mix_mat  = ALLOC_N (int, 40 * ny);
```

Comment:
Any idea where the hard-coded `40` is coming from? In the case statement above, `ny` is assigned `40` but we're hard-coding `40` into the non-`ny` term inside these `ALLOC_N()` invocations.

I did a quick check plotting #mats for `mat` variable from `rect2d.silo` and thresholding on 1.5...max and count 518 zones where mixing is occuring for the default `size` case (30 x 40). In this case `40 * ny` gives 1600 and that is more than enough for the 518 mixing entries. So, I think the code is ok. Its just odd where that `40` comes from.

Reply:
The 40 was just choosen to hopefully over allocated the array. I added a comment to the code.

Reply:
I was thinking that as well 😄 Was just wondering why `40` was a better way to achieve that than `nx`? No worries though.

Reply:
I guess what I mean is, while `40` needs some explanation (e.g. a comment) I don't think `nx` does.

Reply:
I assume the 40 made sense to me when I wrote the code, but I have no idea now, so the comment only mentions that it was chosen to hopefully over allocate the array.

Reply:
are we sure it shouldn't be 42?

Reply:
Oh, sheesh. You're right, `42` would be way better than `40` 😆